### ISSUE TITLE
Discount After Subtotal 

### ISSUE THREAD

**Issue by 123mahaveer** [jekkos/opensourcepos#911]
### New Feature or Enhacement

It would be helpful to add an option for discount after the subtotal, right now the discounts are item specific. 
There are many scenarios where we need to give round off discount of a nett discount on the total bill. Right now there is no option to do this ,it has to be done item-wise only. 

Also is there any indirect work around to do this or is this planned in near future ?

**Comment by daN4cat** [comment]
Have you checked `Default Sales Discount %` functionality in Store Config & General tab?

**Comment by nenuchand** [comment]
Yes i have checked the Default Sales Discount%, that applies the discount on the individual item . Not on the grand total .
For Example for a  total sale of 222.50 i want to give a round off / discount of 2.50 to make the total as 220,that is currently not possible .

**Comment by odiea** [comment]
Maybe its time to remove the % from the line item and add it to the Register. That could possibly help out the other issue where a customer is selected after an item has already been entered into the sale.

**Comment by daN4cat** [comment]
@odiea it wouldn't work for my case, my wife discounts items in different ways sometime. So the overall discount is a different concept. I thought the Default Sales Discount was on the total.

**Comment by odiea** [comment]
It appears the default sales discount is on the item or items in the sale register just like the customer discount.

**Comment by daN4cat** [comment]
Yep, many different discount schema see also #261 #286

**Comment by nenuchand** [comment]
So , we need to code our own solution for the same or is there any workaround ?

**Comment by daN4cat** [comment]
Add the feature to OSPOS, contributions are welcome.

**Comment by deepshah** [comment]
What is the status on this? I am planning to add this, along with a similar feature for taxes. Currently if a tax value changes, or a new tax is added, the same needs to be changed for all items.

**Comment by daN4cat** [comment]
You are welcome to work on this and contribute.
As you can see from the labels it needs an owner.

**Comment by jekkos** [comment]
@deepshah Yes this would be nice to have as a contribution. We still have some PR's open and will try to look into those coming days.

**Comment by Raoufasker** [comment]
please add Feature make client can loop the discount on the total price such as make discount 20% result total price after discount appear then discount again on total result after first discount and make it again and again

**Comment by odiea** [comment]
Here is a simple solution to add a Total discount at the Register.
application/models/sale/Sale.php and add the following at line 977 above check.
$payments[$this->lang->line('customers_discount')] = $this->lang->line('customers_discount');
This will work only for a set amount discount.

**Comment by SteveIreland** [comment]
How would this affect VAT tax reporting?   For sales tax calculations the sales taxes are calculated based on what is being sold.  Won't an invoice level discount mean that we would be over-collecting on taxes?  That doesn't sound right.  For VAT taxes you obviously wouldn't be over-collecting, but you might be over-reporting.

**Comment by odiea** [comment]
And do you think anyone really paid attention to this joke.

**Comment by SteveIreland** [comment]
Wait ... this was a joke?

**Comment by daN4cat** [comment]
Yep the implications of fix discount and tax per item is not a trivial one. Discounting the subtotal however implies that the total tax is calculated against the subtotal but the break down by item needs to be cascaded proportionally.

**Comment by mckaygerhard** [comment]
maybe the solution of @odiea not affecting the reports by a flag in the database.. so reports can take in consideration changes when this flags was present as a overall discount

**Comment by odiea** [comment]
The biggest issue with this Enhancement is some want an additional  % discount while others want a fixed amount discount. This appears to be a no win situation for the poor person wanting to create this feature.

**Comment by mckaygerhard** [comment]
wow @odiea you have been so dramatic, i have a tear on my face!

umm ok i understand now, so why not a option to make it as percentage and also as amount? then everybody wins here!

**Comment by daN4cat** [comment]
Happy for you to implement it. Looking forward to merging the contribution.

**Comment by odiea** [comment]
Here is my contribution. Create a Customer discount Item with the proper tax. Add it to the Sale with a minus sign. I think the amount with the proper tax would be deducted.
![2017-12-17 13_49_25-wedidit _ powered by ospos 3 2 0](https://user-images.githubusercontent.com/10962177/34083974-4b588dbc-e336-11e7-8562-a9735762b687.png)

**Comment by mckaygerhard** [comment]
jajajaj great @odiea was quickly and efective!
